vms/platformvm/vm.go

```diff
@@ -219,6 +219,19 @@ func (vm *VM) Initialize(
 		return err
 	}
 
+	shouldPrune, err := vm.state.ShouldPrune()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to check if the database should be pruned: %w",
+			err,
+		)
+	}
+	if !shouldPrune {
+		chainCtx.Log.Info("state already pruned and indexed")
+		vm.pruned.Set(true)
```

Comment:
By setting `pruned` here - we guaranteed that we will return `nil` from `VerifyHeightIndex` if the P-chain has already been pruned (even if it is called immediately after `Initialize`).